app/src/main/java/com/github/sdp/mediato/SearchBar.java

```diff
@@ -0,0 +1,133 @@
+package com.github.sdp.mediato;
+
+import android.graphics.Paint;
+import android.graphics.Typeface;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ListView;
+import android.widget.SearchView;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+public class SearchBar extends AppCompatActivity implements View.OnClickListener, SearchView.OnQueryTextListener {
+
+    @Override
+    public boolean onQueryTextSubmit(String s) {
+
+        if(s.length() > 0){
+            // this.mTextView.setText("Results");
+            search(s);
+            // toDO : call the search function
+            // toDO : update the ViewList
+        }
+        return false;
+    }
+
+    @Override
+    public boolean onQueryTextChange(String s) {
+        if(s.length() > 0){
+            // this.mTextView.setText("Suggested");
+            search(s);
+            // toDO : call the search function
+            // toDO : update the ViewList
+        }
+        return false;
+    }
+
+    @Override
+    public void onClick(View view) {
+
+        if (view == mPeopleButton) {
+            this.currentCategorie = SearchCategorie.PEOPLE;
+        } else if (view == mBooksButton) {
+            this.currentCategorie = SearchCategorie.BOOKS;
+        } else if (view == mFilmButton) {
+            this.currentCategorie = SearchCategorie.MOVIES;
+        } else if (view == mMusicButton) {
+            this.currentCategorie = SearchCategorie.MUSIC;
+        }
+
+        // toDO : implement a try catch version of this (in case hte currentHighLightedButton is null for example)
+        this.currentHighlightedButton.setTypeface(null, Typeface.NORMAL);
+        this.currentHighlightedButton.setPaintFlags(this.currentHighlightedButton.getPaintFlags() & (~Paint.UNDERLINE_TEXT_FLAG));
+        this.currentHighlightedButton = (Button) view;
+        this.currentHighlightedButton.setTypeface(null, Typeface.BOLD);
+        this.currentHighlightedButton.setPaintFlags(this.currentHighlightedButton.getPaintFlags() | Paint.UNDERLINE_TEXT_FLAG);
+    }
+
+    /* attributes */
+    private enum SearchCategorie{
+        PEOPLE,
+        MOVIES,
+        BOOKS,
+        MUSIC
+    }
+
+    private SearchView mSearchView;
```

Comment:
Is there a reason for the _m_ in front of the field names? Also I think it would be cleaner to have the fields be at the top of the class.

Reply:
to name the attributes starting with an m is a convention I took following an android application bootcamp. But I removed it in the code update as we didn't discussed this convention with the team